### DISCUSSION TITLE
WebSocket in cluster mode

### DISCUSSION THREAD

**Discussion by vincar** [amberframework/amber#1287] [General]
### Description

I see ClientSocket.broadcast is just call channel.rebroadcast!

```
protected def rebroadcast!(message)
    subscribers = ClientSockets.get_subscribers_for_topic(message["topic"])
    subscribers.each_value(&.socket.send(message.to_json))
end
```

when it runs in cluster mode, the socket in other process which subscribe the same topic can not receive the message

**Comment by crimson-knight** [comment]
@vincar I think that makes sense, especially if all running app instances are on the same machine and using the same port. Only a single instance is going to receive the message.

Are you trying to use the `ClientSocket` as a pub/sub system like RabbitMQ or Kafka?
